### DISCUSSION TITLE
DevTools Iphone

### DISCUSSION THREAD

**Discussion by augustinanders** [neuefische/questions#582] [Web] [answered]
I have a Styling Problem that only occurs on my iPhone, but not in the preview on my mac. So I thought it would be helpful to use Devtools to give some test-data to my localstorage. 

Is there a simple way to use Dev-Tools on iPhone?

**Comment by CJAkkuec** [comment]
Have you checked whether that's a browser issue rather than a device issue? You may be using Safari on your iPhone while working with Firefox on your Mac. If there's no browser disparity (because you may be testing in Safari already), then it might be a mobile browser specific issue. Depending on your mobile browser, there's a couple different ways, such as [this](https://www.dummies.com/article/technology/programming-web-design/general-programming-web-design/how-to-use-developer-tools-in-safari-on-ios-205596/) one, if you're using Safari (which is the default mobile browser for iPhones). That aside, mobile browsers on iOS devices are a bit of a dodgy issue, because Apple doesn't love third party browser engines.

**Comment by pixelass** [comment, marked as answer]
If you install x-code you can load the "simulator". You can then choose to keep it in your dock for quick access. The Simulator will launch a virtual but identical environment (you can even switch the version of t OS or a different device)

You can inspect it with the WebInspector from Safari (See @CJAkkuec's response)

> FYI. If it's a bug in mobile safari then the bug will most likely also happen in Safari Desktop (feel free to test it there. All browsers on iOS use safari under the hood (even Chrome or Firefox) because it is the only engine allowed by Apple.

## Finding the simulator
![xcode_001](https://user-images.githubusercontent.com/1148334/234810704-38f1aad8-a526-4f0d-b82f-5b364b51882e.png)
![x-code_2](https://user-images.githubusercontent.com/1148334/234810723-5c1845cc-d7be-40d8-8e8c-a771459a52a0.png)
![Screenshot 2023-04-27 at 10 49 37](https://user-images.githubusercontent.com/1148334/234810838-7b556a04-88ba-4700-a9ae-d3f65129edbb.png)
![Screenshot 2023-04-27 at 10 52 56](https://user-images.githubusercontent.com/1148334/234811469-22095866-1d53-4e2b-b4ff-92c9c44c9fdb.png)

## Setting up Safari
> Enable developer mode

![safarideveloper](https://user-images.githubusercontent.com/1148334/234812236-3f4fbb59-dd33-4d17-b0b4-9b0924bccb59.png)

The connected devices will show up here
![Screenshot 2023-04-27 at 10 56 28](https://user-images.githubusercontent.com/1148334/234812424-e832f637-d7e0-4e3c-bb27-e87fa85554be.png)

**Reply by augustinanders** [reply]
thanks! this helped a lot :)

**Comment by FreytagDom** [comment]
Hy 
You can download "Web Inspector" from the AppStore on your iPhone.
